### ISSUE TITLE
installation error on centos 7

### ISSUE THREAD

**Issue by ChSM56** [thingsboard/thingsboard#417]
My System is Cento 7 (  has 4GB RAM and 225GB disk storage with Dual core processor configuration)  and worked fine with HSQLDB. I tried to populate a Postgres DB and switch but failed. So, I started all over again and ran into the same problem of failure to populate Demo data.

I have run into the install.sh issue
The step I am at:
/usr/share/thingsboard/bin/install/install.sh --loadDemo

I am looking to use Postgresql as the data store.

https://user-images.githubusercontent.com/29800826/32778728-03d72f96-c961-11e7-94e1-bbb67998a404.PNG

https://user-images.githubusercontent.com/29800826/32778874-9cbcf524-c961-11e7-91b0-dd3757d2e365.PNG

I have postgres working fine from another computer in the LAN I has noticed the following SQL files in data/sql folder.

https://github.com/thingsboard/thingsboard/files/1470642/schema.sql.txt

https://github.com/thingsboard/thingsboard/files/1470643/system-data.sql.txt

My Thingsboard config file for Postgres is
https://user-images.githubusercontent.com/29800826/32778874-9cbcf524-c961-11e7-91b0-dd3757d2e365.PNG

Installation is successful with HSQLDb.

This issue I see is with Postgresql Database demo data population.
Just attaching the HSQLDb screenshot:
https://user-images.githubusercontent.com/29800826/32779356-6e1938e8-c963-11e7-845f-3d8177d86741.png

Thank you very much in advance for your help & support.

**Comment by zhw618** [comment]
I use CentOS 7, too. 
And I installed thingsboard successfully, but using  the rpm file.

Try rpm file~ It's  easy.

download the rpm file, and run:
# yum localinstall  thingsboard-1.3.1.rpm
Done!

at last, modified the config file to pgsql and restart the thingsboard service

**Comment by ChSM56** [comment]
Congratulations.

I am sure it should work. I must be missing something.

It is rpm only that I use. I get the rpm file with wget on local folder.

The Installation command is for yum to do local rpm install.

I will get onto this next Tuesday - 21st Nov 2017.

Thank you

**Comment by mp-loki** [comment]
Hi @ChSM56 !
Can you please run the installation with Postgres config one more time and then attach the `/var/log/thingsboard/install.log` file so we can find out what is the cause of the issue?

Thank you

**Comment by ChSM56** [comment]
Hi mp-loki

Please allow me time till Tuesday as I am in hustle for next couple of days. I had an issue with TB-Gateway that I fixed today.

Thank you so much.

**Comment by ChSM56** [comment]
Hi @mp-loki 

I have collected information as you wanted me while making this effort. I am sorry for the confusion as I was on a travel hustle. However, to gain from the help, and help myself with the time you might find for my issue and just about an Hour was good for me to collect the information, i have come back on this.

Hope you understand.

![image](https://user-images.githubusercontent.com/29800826/32990444-67327af4-cd4f-11e7-8dc0-5b47b29bb5a9.png)

![image](https://user-images.githubusercontent.com/29800826/32990500-38e7f57e-cd50-11e7-97f9-9695943ec42b.png)

Attached is the Installation log file -
[install.log](https://github.com/thingsboard/thingsboard/files/1485516/install.log)

Thank you for your support,

Regards

**Comment by mp-loki** [comment]
Hi @ChSM56 . Unfortunately, the install.log is not very helpful. 
Just to check - have you created a **thingsboard** database on your local PostgreSQL server as described in installation guide? are you able to connect to it using psql (or any other tool) using the credentials that you have provided in thingsboard.yml?

**Comment by ChSM56** [comment]
I did create the database and established access to it from a different
desktop over local area network.

Sorry that I am responding on email.

I had attached that screenshot in my previous post and now away from the
system in travel.

I switched back to HSQLDb and found everything working alright.

Just to know which file has the Log4 settings? May be we start there.

Regards

**Comment by mp-loki** [comment]
The logger settings are in /etc/thingsboard/conf/logback.xml

**Comment by ChSM56** [comment]
Hi @mp-loki 

Here is the screenshot my access to Postgres on Linux Centos 7 system from another Windows 10 desktop GUI tool.

![image](https://user-images.githubusercontent.com/29800826/33056412-971cf444-ceab-11e7-9a6c-832d0bf37f49.png)

The connection parameters screen 

![image](https://user-images.githubusercontent.com/29800826/33056451-be1a3d5e-ceab-11e7-881d-1b061ed7ff77.png)

In another attempt, I ran the SQL file from this Windows 10 system and created all the object and tried to switch the connection but in vain.

Thanks

**Comment by ChSM56** [comment]
I am closing this thread for my preference is to get going by simple process of Install & Configure as a standard approach to software use.

**Comment by ChSM56** [comment]
I am posting this to state further on the reason for my setup with Postgres Install failure and how I **solved**

The Unix user - postgres and DB user postgres had same password containing a character - "@". That was not getting through in the access credentials.

I reinstalled (an unnecessary step, though I realized later) POSTGRES and set up the passwords in simple alphanumeric with NO SPECIAL characters.

I made the thingsboard configuration (.yml) file changes to point to the POSTGRES database thingsboard (this could also be changed I think as it is a parameter in the DB Connection String.
